src/blocks/apt.rs

```diff
@@ -192,3 +198,54 @@ fn get_update_count(updates: &str) -> usize {
 fn has_matching_update(updates: &str, regex: &Regex) -> bool {
     updates.lines().any(|line| regex.is_match(line))
 }
+
+struct ResultBool(Result<bool>);
+
+// Need to implement Default to collect futures
+impl Default for ResultBool {
+    fn default() -> ResultBool {
+        ResultBool(Ok(true))
+    }
+}
+
+async fn get_update_count_ignore_phased(config_path: &str, updates: &str) -> Result<usize> {
+    let non_phased_updates =
+        futures::stream::iter(updates.lines().filter(|line| line.contains("[upgradable")))
+            .filter_map(|line| async move {
+                match is_phased_update(config_path, line).await {
+                    Ok(true) => None,
+                    Ok(false) => Some(ResultBool(Ok(true))),
+                    Err(e) => Some(ResultBool(Err(e))),
+                }
+            })
+            .collect::<Vec<ResultBool>>()
+            .await
+            .into_iter()
+            .map(|ResultBool(v)| v)
+            .collect::<Result<Vec<bool>>>()?;
```

Comment:
I think this is a bit over-engineered. Writing this using a simple `for` loop is simpler, more straight-forward and avoids two unnecessary `.collect`s:

```rs
let mut cnt = 0;

for update_line in updates.lines().filter(|line| line.contains("[upgradable")) {
    if !is_phased_update(config_path, update_line).await? {
        cnt += 1;
    }
}

Ok(cnt)
```

Reply:
Ah yes, derp. Good point.